doma-core/src/main/java/org/seasar/doma/jdbc/criteria/statement/UnifiedSelectStarting.java

```diff
@@ -0,0 +1,946 @@
+    Objects.requireNonNull(propertyMetamodel8);
+    Objects.requireNonNull(propertyMetamodel9);
+    return asNativeSqlSelectStarting()
+        .select(
+            propertyMetamodel1,
+            propertyMetamodel2,
+            propertyMetamodel3,
+            propertyMetamodel4,
+            propertyMetamodel5,
+            propertyMetamodel6,
+            propertyMetamodel7,
+            propertyMetamodel8,
+            propertyMetamodel9);
+  }
+
+  /**
+   * Projects the result as a List of {@link Row} implicitly with the specified properties. The
+   * duplicated rows are NOT removed from the result.
+   *
+   * @param propertyMetamodels the property metamodels to project
+   * @return the result
+   */
+  public SetOperand<Row> select(
+      PropertyMetamodel<?> propertyMetamodel, PropertyMetamodel<?>... propertyMetamodels) {
+    Objects.requireNonNull(propertyMetamodel);
+    Objects.requireNonNull(propertyMetamodels);
+    return asNativeSqlSelectStarting().select(propertyMetamodel, propertyMetamodels);
+  }
+
+  /**
+   * Projects the result as a List of {@link Row} explicitly with the specified properties. The
+   * duplicated rows are NOT removed from the result.
+   *
+   * @param propertyMetamodels the property metamodels to project
+   * @return the result
+   */
+  public SetOperand<Row> selectAsRow(
+      PropertyMetamodel<?> propertyMetamodel, PropertyMetamodel<?>... propertyMetamodels) {
+    Objects.requireNonNull(propertyMetamodel);
+    Objects.requireNonNull(propertyMetamodels);
+    return asNativeSqlSelectStarting().selectAsRow(propertyMetamodel, propertyMetamodels);
+  }
+
+  /**
+   * Projects the result to the entity. The duplicated entities are NOT removed from the result.
+   *
+   * @return the result
+   */
+  public <RESULT> SetOperand<RESULT> selectTo(
+      EntityMetamodel<RESULT> entityMetamodel, PropertyMetamodel<?>... propertyMetamodels) {
+    Objects.requireNonNull(entityMetamodel);
+    Objects.requireNonNull(propertyMetamodels);
+    return asNativeSqlSelectStarting().selectTo(entityMetamodel, propertyMetamodels);
+  }
+
+  @Override
+  public Stream<ENTITY> openStream() {
+    return asNativeSqlSelectStarting().openStream();
+  }
+
+  @Override
+  public <RESULT> RESULT mapStream(Function<Stream<ENTITY>, RESULT> streamMapper) {
+    return asNativeSqlSelectStarting().mapStream(streamMapper);
+  }
+
+  @Override
+  public SetOperationContext<ENTITY> getContext() {
+    return asNativeSqlSelectStarting().getContext();
+  }
+
+  @Override
+  public SetOperator<ENTITY> union(SetOperand<ENTITY> other) {
+    return asNativeSqlSelectStarting().union(other);
+  }
+
+  @Override
+  public SetOperator<ENTITY> unionAll(SetOperand<ENTITY> other) {
+    return asNativeSqlSelectStarting().unionAll(other);
+  }
+
+  @Override
+  protected Command<List<ENTITY>> createCommand() {
+    return asEntityqlSelectStarting().createCommand();
+  }
+
+  private UnifiedSelectTerminal<ENTITY> asUnifiedSelectTerminal() {
```

Comment:
I want to use this method by client-code.
Because I want to prevent the class of the receiver for the associate method changes between the first and subsequent calls.

```kotlin
val query = queryDsl.from(a)
    .leftJoin(b) {
        eq(a.id, b.aId)
    }
val associatedQuery = query
    .associate( /* omission */ ) // receiver class is UnifiedSelectStarting
    .associate( /* omission */ ) // receiver class is UnifiedSelectTerminal
    .associate( /* omission */ ) // receiver class is UnifiedSelectTerminal
```

I accept automatically change to UnifiedSelectTerminal, when call UnifiedSelectStarting#associate method. but, I want manually change receiver.

When decomposed query into a separate method, it would require double implementation through overloading like bellow code.

```kotlin
fun findList1() {
    val query = queryDsl.from(a)
        .leftJoin(b) {
            eq(a.id, b.aId)
        }
        .associateTableB() // use first associateTableA method
}

fun findList2() {
    val query = queryDsl.from(a)
        .leftJoin(b) {
            eq(a.id, b.aId)
        }
        .associateTableA()
        .associateTableB() // use second associateTableA method
}

fun UnifiedSelectStarting<TableA>.associateTableA(): UnifiedSelectTerminal<TableA> {
    return this.associate( /* omission */ )
}
// overload
fun UnifiedSelectTerminal<TableA>.associateTableA(): UnifiedSelectTerminal<TableA> {
    return this.associate( /* omission */ )
}

fun UnifiedSelectStarting<TableB>.associateTableB(): UnifiedSelectTerminal<TableB> {
    return this.associate( /* omission */ )
}
// overload
fun UnifiedSelectTerminal<TableB>.associateTableB(): UnifiedSelectTerminal<TableB> {
    return this.associate( /* omission */ )
}
```

Reply:
In commit [40107f4](https://github.com/domaframework/doma/pull/1203/commits/40107f4914c50159976c60e4e333c413469d02ad), we introduced the `EntityQueryable` and `KEntityQueryable` interfaces to abstract the common elements of `UnifiedSelectStarting` and `UnifiedSelectTerminal`. Do these changes help resolve the issues mentioned above?

Reply:
This abstract receiver class for consistent logic. that is helpful.
thank you.